UpdateDependencies.ps1

```diff
@@ -0,0 +1,125 @@
+﻿#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+function EnsureEnvironmentVariable([string]$envVarName)
+{
+    If ([Environment]::GetEnvironmentVariable($envVarName) -eq $null)
+    {
+        throw "Can't find environment variable '$envVarName'"
+    }
+}
+
+function SetEnvIfDefault([string]$envVarName, [string]$value)
+{
+    If ([Environment]::GetEnvironmentVariable($envVarName) -eq $null)
+    {
+        [Environment]::SetEnvironmentVariable($envVarName, $value)
+    }
+}
+
+EnsureEnvironmentVariable "GITHUB_USER"
+EnsureEnvironmentVariable "GITHUB_EMAIL"
+EnsureEnvironmentVariable "GITHUB_PASSWORD"
+
+SetEnvIfDefault 'COREFX_VERSION_URL' 'https://raw.githubusercontent.com/dotnet/versions/master/dotnet/corefx/master/Latest.txt'
+SetEnvIfDefault 'GITHUB_ORIGIN_OWNER' $env:GITHUB_USER
+SetEnvIfDefault 'GITHUB_UPSTREAM_OWNER' 'dotnet'
+SetEnvIfDefault 'GITHUB_PROJECT' 'corefx'
+SetEnvIfDefault 'GITHUB_UPSTREAM_BRANCH' 'master'
+
+
+$CoreFxLatestVersion = Invoke-WebRequest $env:COREFX_VERSION_URL -UseBasicParsing
+$CoreFxLatestVersion = $CoreFxLatestVersion.ToString().Trim()
+
+# Updates the dir.props file with the latest CoreFX build number
+function UpdateValidDependencyVersionsFile
+{
+    if (!$CoreFxLatestVersion)
+    {
+        Write-Error "Unable to find latest CoreFX version at $env:COREFX_VERSION_URL"
+        return $false
+    }
+
+    $DirPropsPath = "$PSScriptRoot\dir.props"
+    $DirPropsText = [IO.File]::ReadAllText($DirPropsPath)
+
+    $CoreFxRegex = new-object Text.RegularExpressions.Regex '<CoreFxExpectedPrerelease>.*</CoreFxExpectedPrerelease>'
+
+    $DirPropsText = $CoreFxRegex.Replace($DirPropsText, "<CoreFxExpectedPrerelease>$CoreFxLatestVersion</CoreFxExpectedPrerelease>")
+
+    [IO.File]::WriteAllText($DirPropsPath, $DirPropsText)
+    return $true
+}
+
+# Updates all the project.json files with out of date version numbers
+function RunUpdatePackageDependencyVersions
+{
+    cmd /c $PSScriptRoot\build.cmd /t:UpdateInvalidPackageVersions | Out-Host
+
+    return $LASTEXITCODE -eq 0
+}
+
+# Creates a Pull Request for the updated version numbers
+function CreatePullRequest
+{
+    git add . | Out-Host
+
+    $UserName = $env:GITHUB_USER
+    $Email = $env:GITHUB_EMAIL
+    $CommitMessage = "Updating CoreFX dependencies to $CoreFxLatestVersion"
+
+    $env:GIT_COMMITTER_NAME = $UserName
+    $env:GIT_COMMITTER_EMAIL = $Email
+    git commit -m "$CommitMessage" --author "$UserName <$Email>" | Out-Host
```

Comment:
If you add `--porcelain` I think this is as simple as checking to see if there were any lines of output. (But maybe this comment is just to mark it as future work which is fine too.)